tests/bluetooth/host/keys/bt_keys_get_addr/src/main.c

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include "testing_common_defs.h"
+#include <addr.h>
+#include <keys.h>
+#include "host_mocks/print_utils.h"
+#include "mocks/keys_help_utils.h"
+
+/* This LUT contains different combinations of ID and Address pairs */
+const struct id_addr_pair testing_id_addr_pair_lut[] = {
+	{ BT_ADDR_ID_1, BT_ADDR_LE_1 },
+	{ BT_ADDR_ID_1, BT_ADDR_LE_2 },
+	{ BT_ADDR_ID_2, BT_ADDR_LE_1 },
+	{ BT_ADDR_ID_2, BT_ADDR_LE_2 }
+};
+
+/* Global iterator to iterate over the ID, Address pairs LUT */
+static uint8_t params_it;
+
+/* Pointer to the current set of testing parameters */
+struct id_addr_pair const *current_params_vector;
+
+/* This list will hold returned references while filling keys pool */
+struct bt_keys *returned_keys_refs[CONFIG_BT_MAX_PAIRED];
+
+BUILD_ASSERT(ARRAY_SIZE(testing_id_addr_pair_lut) == CONFIG_BT_MAX_PAIRED);
+BUILD_ASSERT(ARRAY_SIZE(testing_id_addr_pair_lut) == ARRAY_SIZE(returned_keys_refs));
+
+/*
+ *  Test list is empty
+ *
+ *  Constraints:
+ *   - Empty list at start up
+ *
+ *  Expected behaviour:
+ *   - List is empty at start up
+ */
+void test_list_is_empty_at_startup(void)
+{
+	zassert_true(check_key_pool_is_empty(),
+		     "List isn't empty, make sure to run this test just after a fresh start");
+}
+
+/*
+ *  Test filling the keys pool with (ID, Address) pairs
+ *
+ *  Constraints:
+ *   - Empty list at startup
+ *
+ *  Expected behaviour:
+ *   - A valid reference is returned by bt_keys_get_addr()
+ *   - ID value matches the one passed to bt_keys_get_addr()
+ *   - Address value matches the one passed to bt_keys_get_addr()
+ */
+void test_get_non_existing_keys(void)
+{
+	struct bt_keys *returned_key;
+	uint8_t id = current_params_vector->id;
+	bt_addr_le_t *addr = current_params_vector->addr;
+
+	returned_key = bt_keys_get_addr(id, addr);
+	/* 'params_it' is incremented before running the test case */
```

Comment:
this should not be needed anymore when you do the vector iteration in the test function itself.